src/model/table.py

```diff
@@ -19,12 +20,37 @@ def get_name(self):
     def get_path(self):
         return self.__path__
 
-    # Will be implemented later in the project
-    def set(self):
-        pass
+    def __get_data_path__(self):
+        return os.path.join(self.__path__, "data")
+        
+    def get_primary_key(self):
+        return self.__table_metadata__.primary_key
+
+    def __get_primary_key_from_path__(self, path):
+        return str(path).replace(self.__path__, '').replace(".json", '').replace("data", '')
 
-    def delete(self):
-        pass
+    def overwrite(self):
+        return self.__table_metadata__.overwrite
+
+    def get_indices(self):
+        return self.__table_metadata__.index_keys
+
+    def set(self, data):
+        primary_key = data.get(self.get_primary_key())
+        existing_row = None
+        if primary_key is not None:
+            existing_row = self.get_by_primary_key(primary_key)
```

Comment:
Row.load_by_primary_key(primary_key)

Reply:
Shouldn’t we check if can_overwrite = true and table exists, we should throw an exception ?

Reply:
no we already check this with "x" in row_serialize if file exists and you can't overwrite it will throw exception